tests/tls_e2e/client_enc/client.cpp

```diff
@@ -397,6 +397,6 @@ OE_SET_ENCLAVE_SGX(
     1,    /* ProductID */
     1,    /* SecurityVersion */
     true, /* Debug */
-    128,  /* NumHeapPages */
+    512,  /* NumHeapPages */
```

Comment:
Different tests had more or less headroom to begin with, and some fare better with the change than others, because snmalloc is size-class based. Other size-classed based allocators will exhibit similar behaviour, depending on the interaction between their size-class choices and the specific allocations made by the program.

I've tried to be precise and to raise the amount 64 pages at a time where needed, but if we are concerned about the amounts are still too large, we can review the details with @mjp41 and see what can be done.

As a datapoint, it may be worth noting that tcmalloc initial alloc is 3 * 1024.